src/config_brain_memory.js

```diff
@@ -188,7 +188,7 @@ Upgrade less: ${strings.upgradeLess}
 };
 
 brain.prepareMemory = function() {
-  Memory.username = Memory.username || _.find(Game.spawns, 'owner').owner.username;
+  Memory.username = Memory.username || _.chain(Game.rooms).map('controller').flatten().filter('my').map('owner.username').first().value();
```

Comment:
we don't need to chain, only a find is enouth. plus using filter as you did will all ways not return usernam until all rooms are looped, chaining or not. use only find allow to stop at first found